### PR TITLE
Uncrustify run script

### PR DESCRIPTION
**Pull request by nrbrook**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

Adds a script to run uncrustify on all source code since the xcode uncrustify plugin no longer works. Simply install uncrustify (homebrew or otherwise) and run the script.

I modified the uncrustify config as 2 of the options produced errors in the latest uncrustify on homebrew for some reason, and it wasn't adding newlines at the end of the files as it was missing the `nl_end_of_file_min` option

Prompted as Xcode gives a warning about missing newline at the end of `Classes/CocoaLumberjack.h`

...

### CODE REVIEW

**Comment by OSSPolice** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any changes to library code should be reflected in the CHANGELOG. Please consider adding a note there about your change.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Copyright is not valid. See our default copyright in all of our files (Classes, Demos and Benchmarking use different formats).</td>
    </tr>
  </tbody>
</table>

Invalid copyright: Benchmarking/BaseNSLogging.m

Invalid copyright: Benchmarking/DynamicLogging.m

Invalid copyright: Benchmarking/PerformanceTesting.h

Invalid copyright: Benchmarking/PerformanceTesting.m

Invalid copyright: Benchmarking/StaticLogging.m

Invalid copyright: Benchmarking/main.m

Invalid copyright: Demos/SQLiteLogger/FMDB/FMDatabase.h

Invalid copyright: Demos/SQLiteLogger/FMDB/FMDatabase.m

Invalid copyright: Demos/SQLiteLogger/FMDB/FMDatabaseAdditions.h

Invalid copyright: Demos/SQLiteLogger/FMDB/FMDatabaseAdditions.m

Invalid copyright: Demos/SQLiteLogger/FMDB/FMResultSet.h

Invalid copyright: Demos/SQLiteLogger/FMDB/FMResultSet.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaAsyncSocket/GCDAsyncSocket.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaAsyncSocket/GCDAsyncSocket.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Categories/DDData.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Categories/DDNumber.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Categories/DDRange.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Categories/DDRange.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPAuthenticationRequest.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPAuthenticationRequest.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPConnection.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPConnection.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPLogging.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPMessage.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPMessage.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPResponse.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPServer.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/HTTPServer.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Mime/MultipartFormDataParser.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Mime/MultipartFormDataParser.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Mime/MultipartMessageHeader.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Mime/MultipartMessageHeader.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Mime/MultipartMessageHeaderField.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Mime/MultipartMessageHeaderField.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPAsyncFileResponse.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPAsyncFileResponse.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPDataResponse.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPDataResponse.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPDynamicFileResponse.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPDynamicFileResponse.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPErrorResponse.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPFileResponse.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPFileResponse.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPRedirectResponse.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/Responses/HTTPRedirectResponse.m

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/WebSocket.h

Invalid copyright: Demos/WebServerIPhone/Vendor/CocoaHTTPServer/WebSocket.m

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Tests/Tests/DDSMocking.m [hunk 1]
```diff
@@ -20,14 +20,15 @@ @implementation DDSMocking
 
 @implementation DDBasicMockArgument
 
-- (instancetype)initWithBlock:(void(^)(id object))block {
+- (instancetype)initWithBlock:(void (^)(id object))block {
```
Comment (lolgear) [inline Tests/Tests/DDSMocking.m]: It doesn't format opening curly brackets in methods, does it?

Comment (nrbrook) [reply Tests/Tests/DDSMocking.m]: This change is the space after void. The configuration for function signature braces is to remove the new line. I have only modified the uncrustify configuration as I specified in the pull request so this change is based on the configuration that was already there but has not been applied in a while.

```
# Newline between function signature and open brace
nl_fdef_brace                           = remove        # string (add/force/ignore/remove)
```

File: Benchmarking/BaseNSLogging.m [hunk 2]
```diff
@@ -13,86 +13,81 @@
...
-		DDLogVerbose(@"%@: SpeedTest0 - %lu", FILENAME, (unsigned long)i);
-	}
+    // Log statements that will not be executed due to log level
+
+    for (NSUInteger i = 0; i < SPEED_TEST_0_COUNT; i++) {
+        DDLogVerbose(@"%@: SpeedTest0 - %lu", FILENAME, (unsigned long)i);
+    }
 }
 
 + (void)speedTest1
```
[the remark this answers is not in the document]
Comment (nrbrook) [reply Benchmarking/BaseNSLogging.m]: I'm not sure why this isn't being fixed...

[the remark this answers is not in the document]
Comment (nrbrook) [reply Benchmarking/BaseNSLogging.m]: https://github.com/uncrustify/uncrustify/issues/2175

**Comment by nrbrook** [comment]
Sure but should probably be a different PR.

File: Benchmarking/BaseNSLogging.m (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrbrook) [reply Benchmarking/BaseNSLogging.m]: Resolved by adding additional uncrustify config option in https://github.com/CocoaLumberjack/CocoaLumberjack/pull/1026/commits/e465e8bd8efb7fc1e38ec7efc3d8915d3a7d305a

File: uncrustify.cfg [hunk 3]
```diff
@@ -269,6 +269,9 @@ nl_for_brace                            = remove        # string (add/force/igno
  
 # Newline between function call and open brace
 nl_fcall_brace                          = remove        # string (add/force/ignore/remove)
+
+# Newline between objective-c method and open brace
+nl_oc_mdef_brace						= remove        # string (add/force/ignore/remove)
```
Comment (lolgear) [inline uncrustify.cfg]: It seems that you have different number of spaces, right? :)

Comment (nrbrook) [reply uncrustify.cfg]: Resolved https://github.com/CocoaLumberjack/CocoaLumberjack/pull/1026/commits/19103c9c5c4991bc1a3473eac010e3af7071c6b6

File: Classes/DDFileLogger.h [hunk 4]
```diff
@@ -157,13 +157,13 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
...
  */
-- (void)didArchiveLogFile:(NSString *)logFilePath NS_SWIFT_NAME(didArchiveLogFile(atPath:));
+- (void)didArchiveLogFile:(NSString *) logFilePath NS_SWIFT_NAME(didArchiveLogFile(atPath:));
 
 /**
  *  Called when the roll action was executed and the log was archived.
  *  Executed on global queue with default priority.
  */
-- (void)didRollAndArchiveLogFile:(NSString *)logFilePath NS_SWIFT_NAME(didRollAndArchiveLogFile(atPath:));
+- (void)didRollAndArchiveLogFile:(NSString *) logFilePath NS_SWIFT_NAME(didRollAndArchiveLogFile(atPath:));
```
Comment (hhanesand) [inline Classes/DDFileLogger.h]: Unneeded space.

File: Classes/DDFileLogger.h [hunk 5]
```diff
@@ -157,13 +157,13 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
 /**
  *  Called when a log file was archived. Executed on global queue with default priority.
  */
-- (void)didArchiveLogFile:(NSString *)logFilePath NS_SWIFT_NAME(didArchiveLogFile(atPath:));
+- (void)didArchiveLogFile:(NSString *) logFilePath NS_SWIFT_NAME(didArchiveLogFile(atPath:));
```
Comment (hhanesand) [inline Classes/DDFileLogger.h]: Unneeded space.

File: Classes/DDFileLogger.h [hunk 6]
```diff
@@ -253,7 +253,7 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
  *
  * You can change it by overriding `newLogFileName` and `isLogFile:` methods.
  **/
-- (BOOL)isLogFile:(NSString *)fileName NS_SWIFT_NAME(isLogFile(withName:));
+- (BOOL)isLogFile:(NSString *) fileName NS_SWIFT_NAME(isLogFile(withName:));
```
Comment (hhanesand) [inline Classes/DDFileLogger.h]: Unneeded space.

File: Classes/DDFileLogger.h [hunk 7]
```diff
@@ -427,14 +427,14 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
  *  You can optionally force the current log file to be rolled with this method.
  *  CompletionBlock will be called on main queue.
  */
-- (void)rollLogFileWithCompletionBlock:(void (^ __nullable)(void))completionBlock
+- (void)rollLogFileWithCompletionBlock      :(void (^ __nullable)(void)) completionBlock
```
Comment (hhanesand) [inline Classes/DDFileLogger.h]: Unneeded space.

File: Classes/DDFileLogger.h [hunk 8]
```diff
@@ -427,14 +427,14 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
...
     NS_SWIFT_NAME(rollLogFile(withCompletion:));
 
 /**
  *  Method is deprecated.
  *  @deprecated Use `rollLogFileWithCompletionBlock:` method instead.
  */
-- (void)rollLogFile __attribute((deprecated));
+- (void) rollLogFile __attribute((deprecated));
```
Comment (hhanesand) [inline Classes/DDFileLogger.h]: Unneeded space.

File: Classes/DDFileLogger.h [hunk 9]
```diff
@@ -490,13 +490,13 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
...
 
-+ (instancetype)logFileWithPath:(NSString *)filePath NS_SWIFT_UNAVAILABLE("Use init(filePath:)");
++ (instancetype)logFileWithPath:(NSString *) filePath NS_SWIFT_UNAVAILABLE("Use init(filePath:)");
 
 - (instancetype)init NS_UNAVAILABLE;
 - (instancetype)initWithFilePath:(NSString *)filePath NS_DESIGNATED_INITIALIZER;
 
 - (void)reset;
-- (void)renameFile:(NSString *)newFileName NS_SWIFT_NAME(renameFile(to:));
+- (void)renameFile:(NSString *) newFileName NS_SWIFT_NAME(renameFile(to:));
```
Comment (hhanesand) [inline Classes/DDFileLogger.h]: Unneeded space.

File: Classes/DDFileLogger.h [hunk 10]
```diff
@@ -490,13 +490,13 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
 
 @property (nonatomic, readwrite) BOOL isArchived;
 
-+ (instancetype)logFileWithPath:(NSString *)filePath NS_SWIFT_UNAVAILABLE("Use init(filePath:)");
++ (instancetype)logFileWithPath:(NSString *) filePath NS_SWIFT_UNAVAILABLE("Use init(filePath:)");
```
Comment (hhanesand) [inline Classes/DDFileLogger.h]: Unneeded space.

File: Classes/DDLog.h [hunk 11]
```diff
@@ -488,12 +488,12 @@ FOUNDATION_EXTERN NSString * __nullable DDExtractFileNameWithoutExtension(const
...
  *  Return all the current loggers
  */
-@property (class, nonatomic, copy, readonly) NSArray<id<DDLogger>> *allLoggers;
+@property (class, nonatomic, copy, readonly) NSArray<id<DDLogger> > *allLoggers;
 
 /**
  *  Return all the current loggers
  */
-@property (nonatomic, copy, readonly) NSArray<id<DDLogger>> *allLoggers;
+@property (nonatomic, copy, readonly) NSArray<id<DDLogger> > *allLoggers;
```
Comment (hhanesand) [inline Classes/DDLog.h]: Everything from this comment and up to line 240 needs to be reverted.

File: Demos/CustomLogLevels/MYLog.h [hunk 12]
```diff
@@ -36,16 +36,16 @@
...
+#define LOG_LEVEL_NOTICE (LOG_FLAG_NOTICE | LOG_LEVEL_WARN)    // 0...001111
+#define LOG_LEVEL_INFO   (LOG_FLAG_INFO   | LOG_LEVEL_NOTICE)  // 0...011111
+#define LOG_LEVEL_DEBUG  (LOG_FLAG_DEBUG  | LOG_LEVEL_INFO)    // 0...111111
+
+#define DDLogFatal(frmt, ...)  LOG_MAYBE(NO,  ddLogLevel, LOG_FLAG_FATAL,  0, nil, __PRETTY_FUNCTION__, frmt, ## __VA_ARGS__)
+#define DDLogError(frmt, ...)  LOG_MAYBE(NO,  ddLogLevel, LOG_FLAG_ERROR,  0, nil, __PRETTY_FUNCTION__, frmt, ## __VA_ARGS__)
+#define DDLogWarn(frmt, ...)   LOG_MAYBE(YES, ddLogLevel, LOG_FLAG_WARN,   0, nil, __PRETTY_FUNCTION__, frmt, ## __VA_ARGS__)
+#define DDLogNotice(frmt, ...) LOG_MAYBE(YES, ddLogLevel, LOG_FLAG_NOTICE, 0, nil, __PRETTY_FUNCTION__, frmt, ## __VA_ARGS__)
+#define DDLogInfo(frmt, ...)   LOG_MAYBE(YES, ddLogLevel, LOG_FLAG_INFO,   0, nil, __PRETTY_FUNCTION__, frmt, ## __VA_ARGS__)
+#define DDLogDebug(frmt, ...)  LOG_MAYBE(YES, ddLogLevel, LOG_FLAG_DEBUG,  0, nil, __PRETTY_FUNCTION__, frmt, ## __VA_ARGS__)
```
Comment (hhanesand) [inline Demos/CustomLogLevels/MYLog.h]: Is this added space ok? I'm unfamiliar with the preprocessor.

File: Demos/CaptureASL/CaptureASL/ViewController.m [hunk 13]
```diff
@@ -8,38 +8,41 @@
...
+
+        case ASL_LEVEL_ERR: level = "ASL_LEVEL_ERR";     break;
+
+        case ASL_LEVEL_WARNING: level = "ASL_LEVEL_WARNING"; break;
+
+        case ASL_LEVEL_NOTICE: level = "ASL_LEVEL_NOTICE";  break;
+
+        case ASL_LEVEL_INFO: level = "ASL_LEVEL_INFO";    break;
+
+        case ASL_LEVEL_DEBUG: level = "ASL_LEVEL_DEBUG";   break;
```
Comment (hhanesand) [inline Demos/CaptureASL/CaptureASL/ViewController.m]: I see no reason to introduce newlines to all the switch statements. They are already quite easy to read. This applies to all other switch statements too.

Comment (lolgear) [reply Demos/CaptureASL/CaptureASL/ViewController.m]: @hhanesand 
Here I strongly suggest pure functional style.

```objective-c
- (char *)levelFromTag:(NSInteger)tag {
    switch (tag) {
        case ASL_LEVEL_EMERG   : return "ASL_LEVEL_EMERG";
        case ASL_LEVEL_ALERT   : return "ASL_LEVEL_ALERT";
        case ASL_LEVEL_CRIT    : return "ASL_LEVEL_CRIT";
        case ASL_LEVEL_ERR     : return "ASL_LEVEL_ERR";
        case ASL_LEVEL_WARNING : return "ASL_LEVEL_WARNING";
        case ASL_LEVEL_NOTICE  : return "ASL_LEVEL_NOTICE";
        case ASL_LEVEL_INFO    : return "ASL_LEVEL_INFO";
        case ASL_LEVEL_DEBUG   : return "ASL_LEVEL_DEBUG";
        default                : return NULL;
    }
}
```

**Comment by hhanesand** [comment]
We're in agreement that warnings are frustrating, so we will happily accept a PR that fixes the lack newlines at the end of files. Only issue here was that Uncrustify made a lot of extra changes.

> The configuration can be modified to perhaps reduce the number of changes it makes if some are not desirable.

Yes, I think that is what is needed. Looks like it will require some updating. @lolgear has already created an issue mentioning the fact that we need to update our code formatting standards, which fits well with this. I guess we'll keep this PR open if any @CocoaLumberjack/collaborators want to update the config file.

**Comment by bpoplauschi** [comment]
We should just update the config for uncrustify - doesn't seem to be a big task

**Pull request closed without merging**